### PR TITLE
Enable `fail fast` CI feature to reduce the End-to-End testing consumption

### PR DESCRIPTION
**Pull request by electrocucaracha**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
The current test-infra End-to-End testing behavior continues executing tests until the end. In some cases, this could extend the testing time and block or misuse the CI resources. This change enables a feature to stop the End-to-End execution if a failure is previously raised.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by arora-sagar** [comment]
@electrocucaracha To test this don't we need new porch version. With old version it will always fail and will have timeout. As porch is not able to find the sub-directories.

**Comment by lapentad** [comment]
@electrocucaracha nice catch on the free5gc reconciler!
005.sh passed and 006.sh failed I wonder if it's a timing issue

**Comment by electrocucaracha** [comment]
> @electrocucaracha nice catch on the free5gc reconciler! 005.sh passed and 006.sh failed I wonder if it's a timing issue

I'm experimenting with different VM flavors to see if there is an improvement on provisioning, but in the meantime we need to merge the two PRs in the Catalog repo

File: .prow.yaml [hunk 1]
```diff
@@ -135,7 +135,7 @@ presubmits:
             - "-c"
             - |
               set -eE; cd "$(git rev-parse --show-toplevel)/e2e/terraform"; trap 'terraform destroy -target module.gcp-ubuntu-focal -auto-approve' EXIT;
-              terraform init && timeout 75m terraform apply -target module.gcp-ubuntu-focal -var="pkg_version=v1.0.1" -auto-approve
+              terraform init && timeout 90m terraform apply -target module.gcp-ubuntu-focal -var="fail_fast=true" -auto-approve
```
Comment (arora-sagar) [inline .prow.yaml:138]: @electrocucaracha I recommend increasing, even more, the time, I think 120m should be good as we discussed.

**Pull request merged**